nvidia.go

```diff
@@ -0,0 +1,116 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const ldconfPath = "/etc/ld.so.conf"
+
+func setupNvidiaDriver(containerRootfs string) (err error) {
+	bins, libs, err := getNvidiaFiles()
+	if err != nil {
+		return
+	}
+
+	// mount driver binaries
+	for _, binFile := range bins {
+		if err = mount(binFile, containerRootfs+binFile, "bind", unix.MS_BIND|unix.MS_REC, "rbind"); err != nil {
+			return
+		}
+	}
+
+	// mount driver libraries
+	libDirsMap := make(map[string]interface{})
+	for _, libFile := range libs {
+		if err = mount(libFile, containerRootfs+libFile, "bind", unix.MS_BIND|unix.MS_REC, "rbind"); err != nil {
+			return
+		}
+		libDir := filepath.Dir(libFile)
+		libDirsMap[libDir] = nil
+	}
+
+	// configure dynamic linker run-time bindings
+	ldArgs := []string{"-r", containerRootfs}
+	for dir := range libDirsMap {
+		ldArgs = append(ldArgs, containerRootfs+dir)
+	}
+	ldCmd := exec.Command("ldconfig", ldArgs...)
```

Comment:
I'm curious, why the agent should be aware of  this new libraries?

Reply:
ldconfig comand as follow:
`ldconfig -r  path/to/rootfs  path/to/rootfs/usr/lib64  path/to/rootfs/usr/lib64/vdpau ...(lib dirs)`
Use -r to change to and use containers's rootfs as root directory.
Specified the directories(`filepath.Join(containerRootfs, dir`)  on the command to be processed.